executor/executor.go

```diff
@@ -1,32 +1,78 @@
 package executor
 
 import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/screwdriver-cd/sd-cmd/config"
+	"github.com/screwdriver-cd/sd-cmd/logger"
 	"github.com/screwdriver-cd/sd-cmd/screwdriver/api"
 	"github.com/screwdriver-cd/sd-cmd/util"
 )
 
+var lgr *logger.Logger
+
 // Executor is a Executor endpoint
 type Executor interface {
-	Run() ([]byte, error)
+	Run() error
+}
+
+func prepareLog(namespace, name, version string) (err error) {
+	dirPath := filepath.Join(config.SDArtifactsDir, ".sd", "commands", namespace, name, version)
+	filename := fmt.Sprintf("%v.log", time.Now().Unix())
+	lgr, err = logger.New(dirPath, filename, log.LstdFlags, false)
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
 // New returns each format type of Executor
 func New(sdAPI api.API, args []string) (Executor, error) {
-	ns, cmd, ver, itr, err := util.SplitCmdWithSearch(args)
+	ns, name, ver, itr, err := util.SplitCmdWithSearch(args)
```

Comment:
The tupple(`ns`, `name`, `ver`) is sometimes appear bellow, and the number of return value for `util. SplitCmdWithSearch ` is a litten too much. I think it is good to use a structure for them. Can you reuse `api.Command` structure for it?

And I think `pos` or `offset` is prefer than `itr` for the name.

Reply:
That would be good idea! I will fix.